src/core_plugins/kibana/server/lib/export/collect_panels.js

```diff
@@ -0,0 +1,32 @@
+import { get } from 'lodash';
+
+import collectIndexPatterns from './collect_index_patterns';
+import collectSearchSources from './collect_search_sources';
+
+export const deps = {
+  collectSearchSources,
+  collectIndexPatterns
+};
+
+export default function collectPanels(savedObjectsClient, dashboard) {
+  let panels;
+  try {
+    panels = JSON.parse(get(dashboard, 'attributes.panelsJSON', '[]'));
+    if (!panels) throw new Error('No panels found');
```

Comment:
Throwing here means this function will return a promise sometimes and other times throw an error. I suggest `async` functions 😄